legacy/core/src/main/java/com/fsck/k9/preferences/SettingsImporter.kt

```diff
@@ -46,7 +46,11 @@ class SettingsImporter internal constructor(
                 )
             }
 
-            // TODO: throw exception if neither global settings nor account settings could be found
+            val hasAccountsWithSettings = imported.accounts.any { it.settings != null }
+            if (!globalSettings && !hasAccountsWithSettings) {
```

Comment:
The intention was to deal with an "empty" settings file here so the UI won't display an empty list of items. Please modify the check accordingly.